fdm-calculator/src/balance/organic-matter/degradation.test.ts

```diff
@@ -0,0 +1,159 @@
+        }
+        const cultivations: FieldInput["cultivations"] = [
+            {
+                b_lu_catalogue: "maize",
+            },
+        ] as FieldInput["cultivations"]
+
+        const degradation = calculateOrganicMatterDegradation(
+            soilAnalysis,
+            cultivations,
+            mockCultivationDetailsMap,
+            mockTimeFrame,
+        )
+
+        expect(degradation.total).toBeInstanceOf(Decimal)
+        expect(degradation.total.toNumber()).toBeLessThan(0)
+    })
+
+    it("should return 0 degradation if formula result is negative", () => {
+        // This would happen with a very high SOM, which is unlikely but good to test the floor.
+        const soilAnalysis: SoilAnalysisPicked = {
+            a_som_loi: 100, // unrealistically high
+            a_density_sa: 1.5,
+        }
+        const cultivations: FieldInput["cultivations"] = [
+            {
+                b_lu_catalogue: "maize",
+            },
+        ] as FieldInput["cultivations"]
+
+        const degradation = calculateOrganicMatterDegradation(
+            soilAnalysis,
+            cultivations,
+            mockCultivationDetailsMap,
+            mockTimeFrame,
+        )
+
+        expect(degradation.total.isZero()).toBe(true)
+    })
+
+    it("should cap the annual degradation at 3500 kg/ha", () => {
+        const soilAnalysis: SoilAnalysisPicked = {
+            a_som_loi: 45,
+            a_density_sa: 1.0,
+        }
+        const cultivations: FieldInput["cultivations"] = [
+            {
+                b_lu_catalogue: "maize", // Arable
+            },
+        ] as FieldInput["cultivations"]
+
+        const degradation = calculateOrganicMatterDegradation(
+            soilAnalysis,
+            cultivations,
+            mockCultivationDetailsMap,
+            mockTimeFrame,
+        )
+
+        expect(degradation.total.toNumber()).toBe(-3500)
+    })
+
+    it("should handle multiple years correctly", () => {
+        const multiYearTimeFrame = {
+            start: new Date("2021-01-01"),
+            end: new Date("2023-12-31"), // 3 years
+        }
+        const soilAnalysis: SoilAnalysisPicked = {
+            a_som_loi: 3,
+            a_density_sa: 1.4,
+        }
+        const cultivations: FieldInput["cultivations"] = [
+            {
+                b_lu_catalogue: "maize",
+            },
+        ] as FieldInput["cultivations"]
+
+        const degradationSingleYear = calculateOrganicMatterDegradation(
+            soilAnalysis,
+            cultivations,
+            mockCultivationDetailsMap,
+            mockTimeFrame,
+        )
+        const degradationMultiYear = calculateOrganicMatterDegradation(
+            soilAnalysis,
+            cultivations,
+            mockCultivationDetailsMap,
+            multiYearTimeFrame,
+        )
+
+        // Expect degradation over 3 years to be 3 times the degradation of 1 year.
+        expect(degradationMultiYear.total.toNumber()).toBeCloseTo(
```

Comment:
why to you take the sum and not the average?

Reply:
The unit is kg / ha not kg/ha / yr. Otherwise the supply has also to be averaged first. However in fdm-app we have currently only single year timeframes